src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -207,36 +198,57 @@ def acr_update_set(client,
     registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
 
     if registry.sku.tier == SkuTier.managed.value:
-        if parameters.sku is not None:
-            validate_sku_update(parameters.sku)
+        validate_sku_update(registry.sku.name, parameters.sku)
         if parameters.storage_account is not None:
             parameters.storage_account = None
             logger.warning(
                 "The registry '%s' in '%s' SKU is a managed registry. The specified storage account will be ignored.",
                 registry_name, registry.sku.name)
-        client = get_acr_service_client(MANAGED_REGISTRY_API_VERSION).registries
-    elif registry.sku.tier == SkuTier.basic.value:
-        if hasattr(parameters, 'sku') and parameters.sku is not None:
-            parameters.sku = None
-            logger.warning(
-                "Updating SKU is not supported for registries in Basic SKU. The specified SKU will be ignored.")
-        if parameters.storage_account is not None:
-            parameters.storage_account = ensure_storage_account_parameter(parameters.storage_account)
+    elif registry.sku.tier == SkuTier.classic.value:
+        validate_sku_update(registry.sku.name, parameters.sku)
 
     return client.update(resource_group_name, registry_name, parameters)
 
 
+def _try_aad_token_login(login_server):
+    try:
+        username = "00000000-0000-0000-0000-000000000000"
+        password = get_login_refresh_token(login_server)
+        return username, password
+    except CLIError as e:
+        logger.warning("AAD authentication failed with message: %s", str(e))
+        return None, None
+
+
+def _try_admin_user_login(registry_name):
+    try:
+        cred = acr_credential_show(registry_name)
+        username = cred.username
+        password = cred.passwords[0].value
+        return username, password
+    except CLIError as e:
+        logger.warning("Admin user authentication failed with message: %s", str(e))
+        return None, None
+
+
 def acr_login(registry_name, resource_group_name=None, username=None, password=None):
     """Login to a container registry through Docker.
     :param str registry_name: The name of container registry
     :param str resource_group_name: The name of resource group
     :param str username: The username used to log into the container registry
     :param str password: The password used to log into the container registry
     """
+    from subprocess import PIPE, Popen
+    docker_not_available = "Please verify whether docker is installed and running properly"
+
     try:
-        call(["docker", "ps"], stdout=PIPE, stderr=PIPE)
+        p = Popen(["docker", "ps"], stdout=PIPE, stderr=PIPE)
+        returncode = p.wait()
     except:
```

Comment:
Specify the type of the error. https://docs.python.org/3/library/subprocess.html#exceptions